app/lib/runtime/message-parser.ts

```diff
@@ -95,6 +95,16 @@ export class StreamingMessageParser {
             let content = currentAction.content.trim();
 
             if ('type' in currentAction && currentAction.type === 'file') {
+              // Remove markdown code block syntax if present and file is not markdown
+              if (!currentAction.filePath.endsWith('.md')) {
+                const codeBlockRegex = /^\s*```\w*\n([\s\S]*?)\n\s*```\s*$/;
+                const match = content.match(codeBlockRegex);
+
+                if (match) {
+                  content = match[1].replace(/^[ ]{4}/gm, '').trim(); // Remove common leading 4-space indent
```

Comment:
I don't think its good idea to remove any indenting manually, cuz not all models use that. the reason its happening is the system prompt has the extra indentation in the examples we can first try and remove that and see if the llms are still adding this.

for python codes this can cause issues if not done correctly

Reply:
Hm, I can remove that, not that indentation breaks something.